script/upgrades/MU04/MU04.sol

```diff
@@ -0,0 +1,494 @@
+        )
+      );
+    }
+  }
+
+  /**
+   * @dev Configure reserve collateral assets and spending ratios
+   */
+  function proposal_configureReserveCollateralAssets() private {
+    address[] memory collateralAssets = Arrays.addresses(celo, bridgedUSDC, bridgedEUROC);
+    uint[] memory spendingRatios = Arrays.uints(1e24 * 0.5, 1e24 * 0.5, 1e24 * 0.5); // TODO check spending ratios
+
+    for (uint i = 0; i < collateralAssets.length; i++) {
+      if (!Reserve(reserveProxy).isCollateralAsset(collateralAssets[i])) {
+        transactions.push(
+          ICeloGovernance.Transaction(
+            0,
+            reserveProxy,
+            abi.encodeWithSelector(Reserve(0).addCollateralAsset.selector, collateralAssets[i])
+          )
+        );
+      }
+    }
+
+    transactions.push(
+      ICeloGovernance.Transaction(
+        0,
+        reserveProxy,
+        abi.encodeWithSelector(
+          Reserve(0).setDailySpendingRatioForCollateralAssets.selector,
+          collateralAssets,
+          spendingRatios
+        )
+      )
+    );
+  }
+
+  /**
+   * @dev Updates reserve tokens to include eXOF
+   */
+  function proposal_updateReserveTokens() private {
+    if (!Reserve(reserveProxy).isToken(eXOFProxy)) {
+      transactions.push(
+        ICeloGovernance.Transaction(0, reserveProxy, abi.encodeWithSelector(Reserve(0).addToken.selector, eXOFProxy))
+      );
+    }
+  }
+
+  /**
+   * @dev Updates reserve spenders
+   */
+  function proposal_updateReserveSpenders() private {
+    // remove outdated reserve spender multisig
+    if (Reserve(reserveProxy).isSpender(oldMainReserveMultisig)) {
+      transactions.push(
+        ICeloGovernance.Transaction(
+          0,
+          reserveProxy,
+          abi.encodeWithSelector(Reserve(0).removeSpender.selector, oldMainReserveMultisig, 0)
+        )
+      );
+    }
+
+    // add new reserve spender multisig
+    if (!Reserve(reserveProxy).isSpender(partialReserveMultisig)) {
+      transactions.push(
+        ICeloGovernance.Transaction(
+          0,
+          reserveProxy,
+          abi.encodeWithSelector(Reserve(0).addSpender.selector, partialReserveMultisig)
+        )
+      );
+    }
+  }
+
+  /**
+   * @dev Updates reserve in broker to use the main reserve
+   */
+  function proposal_updateReserveInBroker() private {
+    if (address(Broker(brokerProxy).reserve()) == partialReserveProxy) {
+      transactions.push(
+        ICeloGovernance.Transaction(0, brokerProxy, abi.encodeWithSelector(Broker(0).setReserve.selector, reserveProxy))
+      );
+    }
+  }
+
+  /**
+   * @dev Updates reserve in BiPoolManager to use the main reserve
+   */
+  function proposal_updateReserveInBiPoolManager() private {
+    if (address(BiPoolManager(biPoolManagerProxy).reserve()) == partialReserveProxy) {
+      transactions.push(
+        ICeloGovernance.Transaction(
+          0,
+          biPoolManagerProxy,
+          abi.encodeWithSelector(Broker(0).setReserve.selector, reserveProxy)
```

Comment:
This works because the function has the same selector, but to remove confusion let's use `BiPoolManager(0).setReserve.selector` here